src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala

```diff
@@ -262,6 +276,12 @@ class WdlStandardLibraryFunctionsType extends WdlFunctions[WdlType] {
     case Success(t @ WdlArrayType(WdlArrayType(wdlType))) :: Nil => Success(t)
     case _ => Failure(new Exception(s"Unexpected transpose target: $params"))
   }
+  def select_first(params: Seq[Try[WdlType]]): Try[WdlType] = extractSingleArgument(params) flatMap {
+    case WdlArrayType(WdlOptionalType(innerType)) => Success(innerType)
```

Comment:
Can we add a case for other types too with a failure like in the value implementation ?
It's still better than having a hard to track scala "match error".